module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/ResultsTest.php

```diff
@@ -107,6 +113,230 @@ public function testGetResultTotal(): void
+                    'facet_fields' => [
+                        'topic_facet' => [
+                            ['Research', 16],
+                            ['Psychotherapy', 8],
+                        ],
+                        'building' => [
+                            ['0/Main/', 11],
+                            ['1/Main/Fiction/', 5],
+                            ['0/Sub/', 2],
+                        ]
+                    ]
+                ]
+            ],
+            $this->getParams(null, $config)
+        );
+
+        // No facets configured:
+        $facets = $results->getFacetList();
+        $this->assertIsArray($facets);
+        $this->assertEmpty($facets);
+
+        // Invalid facet:
+        $facets = $results->getFacetList(['format' => 'Format']);
+        $this->assertIsArray($facets);
+        $this->assertEmpty($facets);
+
+        // Valid facet, no configuration:
+        $facets = $results->getFacetList(['topic_facet' => 'Topic']);
+        $this->assertEquals(
+            [
+                'topic_facet' => [
+                    'label' => 'Topic',
+                    'list' => [
+                        [
+                            'value' => 'Research',
+                            'displayText' => 'Research',
+                            'count' => 16,
+                            'operator' => 'AND',
+                            'isApplied' => false,
+                        ],
+                        [
+                            'value' => 'Psychotherapy',
+                            'displayText' => 'Psychotherapy',
+                            'count' => 8,
+                            'operator' => 'AND',
+                            'isApplied' => false,
+                        ],
+                    ],
+                ]
+            ],
+            $facets
+        );
+
+        // Make it an OR facet:
+        $results->getParams()->addFacet('topic_facet', 'Topic', true);
+        $facets = $results->getFacetList();
+        $this->assertEquals(
+            [
+                'topic_facet' => [
+                    'label' => 'Topic',
+                    'list' => [
+                        [
+                            'value' => 'Research',
+                            'displayText' => 'Research',
+                            'count' => 16,
+                            'operator' => 'OR',
+                            'isApplied' => false,
+                        ],
+                        [
+                            'value' => 'Psychotherapy',
+                            'displayText' => 'Psychotherapy',
+                            'count' => 8,
+                            'operator' => 'OR',
+                            'isApplied' => false,
+                        ],
+                    ],
+                ]
+            ],
+            $facets
+        );
+
+        // Add a filter:
+        $results->getParams()->addFilter('~topic_facet:Research');
```

Comment:
Is it worth testing the addition of other types of filters ("plain" and "NOT")? How about testing applied hierarchical filters?

Reply:
Added a test for plain ("AND") hierarchical facet. "NOT" facets are not reflected here, so not possible to check.